src/main/java/net/dv8tion/jda/internal/entities/emoji/CustomEmojiImpl.java

```diff
@@ -101,7 +101,7 @@ public boolean equals(Object obj)
         if (!(obj instanceof CustomEmoji))
             return false;
         CustomEmoji other = (CustomEmoji) obj;
-        return other.getIdLong() == id;
+        return this.id == other.getIdLong() && getName().equals(other.getName());
```

Comment:
I also noticed that this violates the `hashCode` contract, since `a.equals(b) != (a.hashCode() == b.hashCode())`. It might be better to only compare by id, in both equals and hashCode.

Reply:
Done